lib/backend/sqlite.c

```diff
@@ -584,11 +558,8 @@ static rpmRC sqlite_idxdbIter(dbiIndex dbi, dbiCursor dbc, const char *keyp, siz
 	    dbc->key = sqlite3_column_blob(dbc->stmt, 0);
 	}
 	dbc->keylen = sqlite3_column_bytes(dbc->stmt, 0);
-	if (set) {
-	    dbiCursor kc = dbiCursorInit(dbi, 0);
-	    rc = sqlite_idxdbByKey(dbi, kc, dbc->key, dbc->keylen, set);
-	    dbiCursorFree(dbi, kc);
-	}
+	if (set)
+	    rc = sqlite_idxdbByKey(dbi, dbc->subc, dbc->key, dbc->keylen, set);
```

Comment:
I think there's an error here. It looks like there should be braces here...

Reply:
What makes you think that? There's no need for braces for a single line.

Reply:
Well, aside from a couple lines up there is braces used even with a single line branch on `if`, the `rc = RPMRC_OK` made me think there's some kind of error here.

Reply:
The brace for single line items varies a lot throughout rpm, the only hard rule there is that anything spanning multiple lines in the source must use braces, even if the compiler thinks it's just one line.

I generally dislike unnecessary braces but sometimes they are introduced in anticipation for surrounding code, at other times they get left behind from other code changes, at other times they add to readability and at others, they're just ugly.